db/migrate/20240209120111_add_missing_indexes_to_allocator.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class AddMissingIndexesToAllocator < ActiveRecord::Migration[6.1]
+  def change
+    add_index :allocator, :deleted_at, name: "index_allocator_deleted_at"
+    add_index :allocator, :role_name, name: "index_allocator_role_name"
+    add_index :allocator, :ror_id, name: "index_allocator_ror_id"
```

Comment:
@kaysiz have we confirmed that the queries missing these indices are in fact using these indices in the query execution plan?

Reply:
Good question Wendel!

I think if Rails is not smart enough to have done this and we have to do it manually, it can be a separate piece of work/PR and doesn't need to be a blocker to merging this.